airbyte-integrations/connectors/source-stripe/unit_tests/integration/test_events.py

```diff
@@ -0,0 +1,350 @@
+        start_date = _NOW - timedelta(days=30)
+        lookback_window = timedelta(days=10)
+        http_mocker.get(
+            HttpRequest(
+                url="https://api.stripe.com/v1/events",
+                query_params={"created[gte]": str(int((start_date - lookback_window).timestamp())), "created[lte]": str(int(_NOW.timestamp())), "limit": 100},
+                headers=_AUTHENTICATION_HEADERS,
+            ),
+            _a_response().build()
+        )
+
+        self._read(_config().with_start_date(start_date).with_lookback_window_in_days(lookback_window.days))
+
+        # request matched http_mocker
+
+    @HttpMocker()
+    def test_given_slice_range_when_read_then_request_before_start_date(self, http_mocker: HttpMocker) -> None:
+        start_date = _NOW - timedelta(days=30)
+        slice_range = timedelta(days=20)
+        slice_datetime = start_date + slice_range
+        http_mocker.get(
+            HttpRequest(
+                url="https://api.stripe.com/v1/events",
+                query_params={"created[gte]": str(int(start_date.timestamp())), "created[lte]": str(int(slice_datetime.timestamp())), "limit": 100},
+                headers=_AUTHENTICATION_HEADERS,
+            ),
+            _a_response().build()
+        )
+        http_mocker.get(
+            HttpRequest(
+                url="https://api.stripe.com/v1/events",
+                query_params={"created[gte]": str(int(slice_datetime.timestamp()) + 1), "created[lte]": str(int(_NOW.timestamp())), "limit": 100},
+                headers=_AUTHENTICATION_HEADERS,
+            ),
+            _a_response().build()
+        )
+
+        self._read(_config().with_start_date(start_date).with_slice_range_in_days(slice_range.days))
+
+    @HttpMocker()
+    def test_given_http_status_400_when_read_then_stream_is_ignored(self, http_mocker: HttpMocker) -> None:
+        http_mocker.get(
+            HttpRequest(
+                url="https://api.stripe.com/v1/events",
+                query_params=ANY_QUERY_PARAMS,
+                headers=_AUTHENTICATION_HEADERS
+            ),
+            HttpResponse(
+                json.dumps({
+                    "error": {
+                        "code": "parameter_invalid_integer",
+                        "doc_url": "https://stripe.com/docs/error-codes/parameter-invalid-integer",
+                        "message": "Invalid integer: not_an_integer",
+                        "param": "limit",
+                        "request_log_url": "https://dashboard.stripe.com/test/logs/req_yVDTOo9Jrd1suG?t=1701959063",
+                        "type": "invalid_request_error"
+                    }
+                }),
+                400)
+        )
+        output = self._read(_config().with_start_date(_A_START_DATE))
+        assert len(output.get_stream_statuses(_STREAM_NAME)) == 0
+
+    @HttpMocker()
+    def test_given_http_status_401_when_read_then_stream_is_incomplete(self, http_mocker: HttpMocker) -> None:
```

Comment:
401 (i.e. authentication issues) are still a prevalent cases we face in general for our connectors. I don't know if it's the case per se for stripe. I would assume not because right now, the failure type is `system_error` and not `config_error`. 

My intent was to show that we would be paged given users are unable to log in. Because of [this code block](https://github.com/airbytehq/airbyte/blob/7a9ee9adec0994769f25b6ec24146a5375c41b43/airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py#L135-L140), I assumed the status INCOMPLETE would be sufficient but checking this again, it feels it does not mean the stream will break. Hence, the assertion should be updated to `output.errors[0].trace.error.failure_type`. Does that make sense?

Reply:
> I don't know if it's the case per se for stripe. I would assume not because right now, the failure type is system_error and not config_error.

can we file an issue and add it to the api-connectors team's backlog? we definitely don't want to be paged because 3 users failed to authenticate

Reply:
Added [here](https://github.com/airbytehq/airbyte/issues/33420)